service/kernel/oneapi/cl_kernels/select_indexed.cl

```diff
@@ -46,86 +46,102 @@ void partition( __global  algorithmFPType *pdist,
                 int local_size,
                 algorithmFPType pivot,
                 int *split,
-                int *total_big_num) 
+                int *total_big_num,
+                bool need_init_labels) 
 {
     const int global_id  = get_global_id(0);
```

Comment:
Obvious name. Can we give object-oriented name for this? Also, it confuses that `global_id` is calculated here, but `local_id` is taken as input. 
Suppose this `local_id` is something different from `get_local_id(0)`